great_expectations/cli/suite.py

```diff
@@ -94,20 +103,77 @@ def suite(ctx):
     help="By default launch jupyter notebooks, unless you specify --no-jupyter flag.",
 )
 @click.pass_context
-def suite_new(ctx, expectation_suite, interactive, profile, batch_request, no_jupyter):
+def suite_new(
+    ctx,
+    expectation_suite,
+    interactive_flag,
+    no_interactive_flag,
+    profile,
+    batch_request,
+    no_jupyter,
+):
     """
     Create a new empty Expectation Suite.
     Edit in jupyter notebooks, or skip with the --no-jupyter flag.
     """
     context: DataContext = ctx.obj.data_context
     usage_event_end: str = ctx.obj.usage_event_end
 
+    processed_flags = _process_suite_new_flags_and_prompt(
+        context=context,
+        usage_event_end=usage_event_end,
+        interactive_flag=interactive_flag,
+        no_interactive_flag=no_interactive_flag,
+        profile=profile,
+        batch_request=batch_request,
+    )
+
+    _suite_new_workflow(
+        context=context,
+        expectation_suite_name=expectation_suite,
+        interactive=processed_flags["interactive"],
+        profile=processed_flags["profile"],
+        no_jupyter=no_jupyter,
+        usage_event=usage_event_end,
+        batch_request=batch_request,
+    )
+
+
+def _process_suite_new_flags_and_prompt(
+    context: DataContext,
+    usage_event_end: str,
+    interactive_flag: bool,
+    no_interactive_flag: bool,
+    profile: bool,
+    batch_request: Optional[str] = None,
+) -> Dict[str, Optional[bool]]:
+    """
+    Process various optional suite new flags and prompt if there is not enough information from the flags.
+    Args:
+        context: Data Context for use in sending error messages if any
+        usage_event_end: event name for ending usage stats message
+        interactive_flag: --interactive from the `suite new` CLI command
+        no_interactive_flag: --non-interactive from the `suite new` CLI command
+        profile: --profile from the `suite new` CLI command
+        batch_request: --batch-request from the `suite new` CLI command
+
+    Returns:
+        Dictionary with keys of processed parameters and boolean values e.g.
+        {"interactive": True, "profile": False}
+    """
+
     error_message: Optional[str] = None
 
-    if not interactive and (profile or (batch_request is not None)):
-        error_message = """Using --profile flag and/or --batch-request <path to JSON file> option requires \
---interactive flag.
-"""
+    # Convert interactive / no-interactive flags to interactive
+    interactive: Optional[bool] = None
+    if interactive_flag is True and no_interactive_flag is True:
+        error_message = """Please choose either --interactive or --non-interactive, you may not choose both."""
+    elif interactive_flag is False and no_interactive_flag is False:
+        interactive = None
+    elif interactive_flag is True and no_interactive_flag is False:
+        interactive = True
+    elif interactive_flag is False and no_interactive_flag is True:
```

Comment:
@anthonyburdi Likewise, booleans can be used with "is True" and "is False" -- just a logical expression should be fine; or is there a special intention here?  If so, fine (but curious as to what is the rationale).  Thanks.

Reply:
I was a bit more explicit here because I'm using an `Optional[bool]` for the `interactive` variable (None = no flags present, True = `--interactive` flag present, False = `--non-interactive` flag present). I wanted to be explicit since Python coerces `None` to `False` when you are not explicit. I used them on other boolean comparisons as well just to make sure I did not mess anything up but can clean some of them up.

Reply:
Got it.  I did not realize that the `Optional[bool]` was defaulted to `None` (as opposed to one of `False` / `True`).  If so, then totally fine (though I would truly appreciate a comment for such situations).

Reply:
Good call! I added a comment :)